### PR TITLE
Updates some admin stuff

### PR DESCRIPTION
**Pull request by AffectedArc07**
## What Does This PR Do
Heres the stuff people actually care about

`Camera Range Display` verb is now better
image

Applying bans with `Ban Panel` to `commanddept` and other departments now bans all jobs inside that department

There is now the option to disable auto population of IP and CID when banning live players with the ban panel (IE: If we just want to ban that ckey to make them change it)

Notes and bans now have round IDs attached
image
image

Theres now a button to see karma unlocks
image

Theres a few other misc changes in this PR but they aren't worth mentioning

## Why It's Good For The Game
Better admin tooling

## Changelog
:cl: AffectedArc07
add: Updated ingame admin stuff. 
/:cl:

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Yes I am aware these numbers dont add up
image

I had to mod my DB to force unlocks

File: code/modules/admin/verbs/mapping.dm [hunk 1]
```diff
@@ -56,12 +47,16 @@ GLOBAL_VAR_INIT(intercom_range_display_status, 0)
...
-		qdel(C)
+	for(var/obj/effect/debugging/marker/M in world)
+		qdel(M)
 
 	if(GLOB.camera_range_display_status)
 		for(var/obj/machinery/camera/C in GLOB.cameranet.cameras)
-			new/obj/effect/debugging/camera_range(C.loc)
+			for(var/turf/T in orange(7, C))
+				var/obj/effect/debugging/marker/F = new/obj/effect/debugging/marker(T)
+				if(!(F in view(7, C.loc)))
```
Comment (dearmochi) [inline code/modules/admin/verbs/mapping.dm:57]: Cache the `view(7, C.loc)` out of the `orange(7, C)` loop to avoid redundancy.

**Review by dearmochi — CHANGES_REQUESTED** [review]
Few things

File: code/modules/admin/IsBanned.dm [hunk 2]
```diff
@@ -28,6 +31,7 @@
 	if(!GLOB.guests_allowed && IsGuestKey(key))
 		log_adminwarn("Failed Login: [key] [computer_id] [address] - Guests not allowed")
 		// message_admins("<span class='notice'>Failed Login: [key] - Guests not allowed</span>")
+		INVOKE_ASYNC(GLOBAL_PROC, .proc/log_connection, ckey(key), address, computer_id, CONNECTION_TYPE_DROPPED_BANNED)
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/modules/admin/IsBanned.dm:34]: Connection logs are associated with ckeys so we would also see the ckey field as `Guest-2359832095`

File: code/modules/admin/cookielog.dm [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+// Some of these params can be null, others CAN NOT
+/datum/cookie_record/New(holder_ckey, matched_ckey, matched_ip, matched_cid)
+	// Right off the bat
+	cookie_holder_ckey = holder_ckey
+
+	var/has_note = FALSE
+	var/raw_text = ""
+	// Now lets see if we have a note logging the infraction in the past
+	var/datum/db_query/check_existing_note = SSdbcore.NewQuery("SELECT notetext FROM [format_table_name("notes")] WHERE ckey=:ckey AND adminckey=:ackey", list(
```
[the remark this answers is not in the document]
Comment (AffectedArc07) [reply code/modules/admin/cookielog.dm:38]: There will only ever be one note by this ckey, so LIMIT would not be useful here

File: code/modules/admin/verbs/mapping.dm (hunk 1 above)
Comment (AffectedArc07) [reply code/modules/admin/verbs/mapping.dm:57]: This is a verb used by people on local test servers when mapping and only invokes once (its not a constant ticking loop)

File: code/modules/admin/cookielog.dm (hunk 3 above)
[the remark this answers is not in the document]
Comment (dearmochi) [reply code/modules/admin/cookielog.dm:38]: Ah, I wasn't sure how notes were stored but that makes sense

**Comment by Kiyahitayika** [comment]
I only ever an admin when I'm testing my code and this is just such a nice improvement.

**Review by variableundefined — APPROVED** [review]

**Pull request merged**